blockchain/v2/processor.go

```diff
@@ -137,20 +122,30 @@ func (state *pcState) purgePeer(peerID p2p.ID) {
 // handle processes FSM events
 func (state *pcState) handle(event Event) (Event, error) {
```

Comment:
Code is beautiful, super easy to go through it and understand what's happening. In my view, the only missing piece is link to the spec (ideally TLA+ spec, not English :).